src/code_review_agent/config.py

```diff
@@ -0,0 +1,172 @@
+    
+    # Cache Configuration
+    redis_url: Optional[str] = Field(None, description="Redis URL for caching")
+    cache_ttl_seconds: int = Field(3600, description="Cache TTL in seconds")
+    
+    # File Patterns
+    include_patterns: List[str] = Field(
+        default=[
+            "*.py", "*.js", "*.ts", "*.jsx", "*.tsx", "*.java", "*.go", 
+            "*.rs", "*.cpp", "*.c", "*.h", "*.hpp", "*.cs", "*.php", 
+            "*.rb", "*.swift", "*.kt", "*.scala", "*.clj", "*.sh"
+        ],
+        description="File patterns to include in review"
+    )
+    exclude_patterns: List[str] = Field(
+        default=[
+            "*.min.js", "*.min.css", "node_modules/**", "__pycache__/**",
+            "*.pyc", "*.pyo", "*.pyd", ".git/**", ".venv/**", "venv/**",
+            "build/**", "dist/**", "target/**", "*.lock", "*.log"
+        ],
+        description="File patterns to exclude from review"
+    )
+    
+    class Config:
+        """Pydantic configuration."""
+        env_file = ".env"
+        env_file_encoding = "utf-8"
+        case_sensitive = False
+        
+    @validator("ai_model")
+    def validate_ai_model(cls, v: AIModel, values: dict) -> AIModel:
+        """Validate AI model configuration."""
+        openai_key = values.get("openai_api_key")
+        anthropic_key = values.get("anthropic_api_key")
+        
+        if v.value.startswith("gpt") and not openai_key:
+            raise ValueError("OpenAI API key required for GPT models")
+        elif v.value.startswith("claude") and not anthropic_key:
+            raise ValueError("Anthropic API key required for Claude models")
+            
+        return v
+    
+    @validator("log_file")
+    def validate_log_file(cls, v: Optional[str]) -> Optional[str]:
+        """Ensure log directory exists."""
+        if v:
+            log_path = Path(v)
+            log_path.parent.mkdir(parents=True, exist_ok=True)
+        return v
+    
+    @property
+    def is_development(self) -> bool:
+        """Check if running in development mode."""
+        return self.environment == Environment.DEVELOPMENT
+    
+    @property
+    def is_production(self) -> bool:
+        """Check if running in production mode."""
+        return self.environment == Environment.PRODUCTION
+    
+    @property
+    def has_openai_key(self) -> bool:
+        """Check if OpenAI API key is configured."""
+        return bool(self.openai_api_key)
+    
+    @property
+    def has_anthropic_key(self) -> bool:
+        """Check if Anthropic API key is configured."""
+        return bool(self.anthropic_api_key)
+    
+    @property
+    def uses_openai_model(self) -> bool:
+        """Check if using OpenAI model."""
+        return self.ai_model.value.startswith("gpt")
+    
+    @property
+    def uses_anthropic_model(self) -> bool:
+        """Check if using Anthropic model."""
+        return self.ai_model.value.startswith("claude")
+
+
+# Global settings instance
+settings = Settings()
```

Comment:
Instantiating `Settings()` at import time makes the package import fail if required env vars (e.g., `GITHUB_TOKEN`) aren’t present yet, and prevents late binding during tests/CLI. Prefer a lazy, cached accessor instead of a module-level instance.

```suggestion
from functools import lru_cache

@lru_cache
def get_settings() -> Settings:
    """Get application settings (cached)."""
    return Settings()
```

Then remove the module-level `settings` variable and update call sites to use `get_settings()`.